skimage/feature/peak.py

```diff
@@ -1,22 +1,24 @@
 import numpy as np
 import scipy.ndimage as ndi
 from ..segmentation import relabel_sequential
+from .. import measure
+from ..filters import rank_order
 
 
 def _get_high_intensity_peaks(image, mask, num_peaks):
     """
     Return the highest intensity peak coordinates.
     """
+    # get coordinates of peaks
     coord = np.nonzero(mask)
-
-    intensities = image[coord]
-    sorted_indices = np.argsort(intensities)[::-1]
-
-    # num_peaks might be np.inf, so this check is required
+    # select num_peaks peaks
     if len(coord[0]) > num_peaks:
-        sorted_indices = sorted_indices[:num_peaks]
-
-    return np.transpose(coord)[sorted_indices]
+        intensities = image[coord]
+        idx_maxsort = np.argsort(intensities)
+        coord = np.transpose(coord)[idx_maxsort][-num_peaks:]
```

Comment:
I'm not a user of this function, but does the order of coords matter here (was num_peaks sorted from most to least intensity, now reversed)?